authn/authn.go

```diff
@@ -11,28 +13,37 @@ import (
 	"strings"
 )
 
-// JWTValidator is the interface in charge of extracting JWT claims from request.
-type JWTValidator interface {
-	ValidateRequest(*http.Request) (*Claims, error)
+// UserInfo contains the necessary attributes used in Doorman policies.
+type UserInfo struct {
+	ID     string
+	Email  string
+	Groups []string
 }
 
-var jwtValidators map[string]JWTValidator
+// Authenticator is in charge of authenticating requests.
+type Authenticator interface {
+	ValidateRequest(*http.Request) (*UserInfo, error)
+}
+
+var authenticators map[string]Authenticator
 
 func init() {
-	jwtValidators = map[string]JWTValidator{}
+	authenticators = map[string]Authenticator{}
 }
 
-// NewJWTValidator instantiates or reuses an existing JWT validator for the specified issuer.
-func NewJWTValidator(issuer string) (JWTValidator, error) {
-	if !strings.HasPrefix(issuer, "https://") {
-		return nil, fmt.Errorf("issuer %q not supported or has bad format", issuer)
+// NewAuthenticator instantiates or reuses an existing one for the specified
+// identity provider.
+func NewAuthenticator(idP string) (Authenticator, error) {
+	if !strings.HasPrefix(idP, "https://") {
+		return nil, fmt.Errorf("Identity provider %q not supported or has bad format", idP)
```

Comment:
I think this error message would be better as, `fmt.Errorf("Identify provider %q does not use the https scheme", idP)`. That is clearer what the failure/rejection reason is.